scripts/are_queued_jobs_finished.py

```diff
@@ -68,11 +82,21 @@ def isFinished(jobs):
     if previously_queued_ids:
         current_statuses = getJobStatus(previously_queued_ids)
         finished_jobs = []
-        for job_id, status in current_statuses:
-            if status != 'F':
-                is_finished = False
-                continue
-            finished_jobs.append(job_id)
+        for job in current_statuses.split('\n\n'):
+            if job:
+                job_id = re.search(r'^Job Id: (\d+)\.[\W\w]+$', job).group(1)
+                job_state = re.search(r'job_state = (\w)', job).group(1)
+                if job_state == 'F':
+                    job_status = {'status' :'WAITING', 'color' : 'CYAN' }
+
+                    # if exit status is non-zero, fail the job
+                    if int(re.search(r'Exit_status = (\d+)', job).group(1)):
+                        job_status = { 'status' :'FAIL', 'color' : 'RED' }
+
+                    finished_jobs.extend([job_id, job_status])
```

Comment:
Might be nicer to just have `finished_jobs[job_id] = job_status` so you don't have to do the stride splicing above.